common/lib/xmodule/xmodule/course_metadata_utils.py

```diff
@@ -0,0 +1,159 @@
+"""Simple utility functions that operate on course metadata.
+
+Ths is a place to put simple functions that operate on course metadata. It
+allows us to share code between the CourseDescriptor and CourseOverview
+classes, which both need these type of functions.
+"""
+from datetime import datetime
+from base64 import b32encode
+
+from django.utils.timezone import UTC
+
+from .fields import Date
+
+DEFAULT_START_DATE = datetime(2030, 1, 1, tzinfo=UTC())
+
+
+def clean_id(location, padding_char):
+    """
+    Returns a unique deterministic base32-encoded ID for a course.
+
+    Arguments:
+        location (UsageKey): The UsageKey of said course
+        padding_char (str): Character used for padding at end of the encoded string.
+                            The standard value for this is '='.
+    """
+    return "course_{}".format(
+        b32encode(unicode(location.course_key)).replace('=', padding_char)
+    )
+
+
+def course_url_name(location):
+    """
+    Given a course location, return the course's URL.
+
+    Arguments:
+        location (UsageKey): The usage key of the course in question.
+    """
+    return location.name
+
+
+def display_name_with_default(descriptor):
```

Comment:
Why did you choose to pass in the `descriptor` in this case, whereas for all the other functions you passed in only the required fields?